doc/kubernetes/modules/ROOT/pages/running/metrics/keycloak_service_level_indicators.adoc

```diff
@@ -0,0 +1,160 @@
+| Use the Prometheus `up` metric which indicates if the Prometheus server is able to scrape metrics from the {project_name} instance.
+
+| Latency
+| Response time for authentication related HTTP requests as measured by the server
+| 95% of all authentication related requests should be faster than 250 ms within a 5-minute-interval.
+| {project_name} server-side metrics to track latency for specific endpoints along with Response Time Distribution using `http_server_requests_seconds_bucket` and `http_server_requests_seconds_count`.
+
+| Errors
+| Failed authentication requests due to server problems as measured by the server
+| The rate of errors due to server problems for authentication requests should be less than 0.1% within a 5-minute-interval.
+| Identify server side error by filtering the metric `http_server_requests_seconds_count` on the tag `outcome` for value `SERVER_ERROR`.
+
+|===
+
+^*^ These SLO target values are an example and should be tailored to fit your use case and deployment.
+
+== PromQL queries
+
+=== Availability
+
+This metric will have a value of at least one if the {project_name} instances is available
+and responding to Prometheus scrape requests,
+and 0 if the service is down or unreachable.
+
+Then use a tool like Grafana to show a 30-day interval and let it calculate the average of the metric in that time window.
+
+----
+sum(
+  up{
+    container="keycloak", # <1>
+    namespace="$namespace"
+  }
+)
+OR
+on() vector(0) # <2>
+----
+<1> Filter by additional tags to identify Keycloak
+<2> Alternative value 0 when none of the Pods is available
+
+=== Latency of authentication requests
+
+This Prometheus query calculates the percentage of authentication requests
+that completed within 0.25 seconds relative to all authentication requests for specific Keycloak endpoints, targeting a particular namespace and pod, over the past 5 minutes.
+
+This example requires the Keycloak configuration `http-metrics-slos` to be set to `250` indicating that buckets for requests faster and slower than 250 ms should be recorded.
+Setting `http-metrics-histograms-enabled` to `true` would capture additional buckets which can help with performance troubleshooting.
+
+----
+sum(
+  rate(
+    http_server_requests_seconds_bucket{
+      uri=~"/realms/{realm}/protocol/{protocol}/.*|/realms/{realm}/login-actions/.*", # <1>
+      le="0.25", # <2>
+      container="keycloak", # <3>
+      namespace="$namespace"}
+    [5m] # <4>
+  )
+) without (le,uri,status,outcome,method,pod,instance) # <5>
+/
+sum(
+  rate(
+    http_server_requests_seconds_count{
+      uri=~"/realms/{realm}/protocol/{protocol}/.*|/realms/{realm}/login-actions/.*", # <1>
+      container="keycloak",
+      namespace="$namespace"}
+    [5m] # <3>
+  )
+) without (le,uri,status,outcome,method,pod,instance) # <5>
+----
+<1> URLs related to logging in
+<2> Response time as defined by SLO
+<3> Filter by additional tags like
```

Comment:
Does the sentence look unfinished to you? @ahus1

Reply:
fixed in next commit